### PR TITLE
pwm_out_sim cleanup

### PR DESCRIPTION
**Pull request by dagar**
One of the intermittent CI mission test failures appears to be a problem with pwm_out_sim. While debugging I did a bit of a cleanup.

 - move to ModuleBase
 - strip down to PWM 8 and 16 modes only
 - remove all dead code
 - implement missing pwm ioctls (current value, rates, etc)
 - default rate 50Hz -> 400Hz

### CODE REVIEW

File: src/drivers/pwm_out_sim/PWMSim.hpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef DRIVERS_PWM_OUT_SIM_PWMSIM_HPP_
+#define DRIVERS_PWM_OUT_SIM_PWMSIM_HPP_
+
+/**
+ * @file hil.cpp
```
Comment (LorenzMeier) [inline src/drivers/pwm_out_sim/PWMSim.hpp]: Needs update for correct file.

**Comment by dagar** [comment]
I'll test HIL quickly.

**Comment by dagar** [comment]
jMavsim HITL still working.

https://logs.px4.io/plot_app?log=c27ae0a0-b333-4948-99f8-db0673bf3364

**Comment by dagar** [comment]
Reducing pwm sim task stack a bit.

image

File: src/drivers/pwm_out_sim/PWMSim.cpp [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+mix them with any loaded mixer and output the result to the
+actuator_output uORB topic. PWMSim can also be performed with normal
+PWM outputs, a special flag prevents the outputs to be operated
+during PWMSim mode. If PWMSim is not performed with a standalone FMU,
+but in a real system, it is NOT recommended to use this virtual
+driver. Use instead the normal FMU or IO driver.
+
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("pwmsim", "driver");
```
Comment (bkueng) [inline src/drivers/pwm_out_sim/PWMSim.cpp]: This needs to match the shell command: `pwm_out_sim` (and the same for `px4_task_spawn_cmd`).

File: src/drivers/pwm_out_sim/PWMSim.cpp [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+	PRINT_MODULE_USAGE_COMMAND_DESCR("start", "Start the task (without any mode set, use any of the mode_* cmds)");
```
Comment (bkueng) [inline src/drivers/pwm_out_sim/PWMSim.cpp]: `without any mode set` is this correct? The constructor initializes with pwm16

File: src/drivers/pwm_out_sim/PWMSim.cpp [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+
+Driver/configurator for the virtual PWMSim port.
+
+This virtual driver emulates PWM / servo outputs for setups where
+the connected hardware does not provide enough or no PWM outputs.
+
+Its only function is to take actuator_control uORB messages,
+mix them with any loaded mixer and output the result to the
+actuator_output uORB topic. PWMSim can also be performed with normal
+PWM outputs, a special flag prevents the outputs to be operated
```
Comment (bkueng) [inline src/drivers/pwm_out_sim/PWMSim.cpp]: Can you document what the special flag is and how it's configured?

**Pull request merged**